src/compute-client/src/controller/instance.rs

```diff
@@ -306,10 +306,51 @@ where
     /// This method removes the replica from the orchestrator and should only be called if the
     /// replica should be permanently removed.
     pub async fn remove_replica(&mut self, id: ReplicaId) -> Result<(), ComputeError> {
+        if !self.compute.replicas.contains_key(&id) {
+            return Err(ComputeError::ReplicaMissing(id));
+        };
```

Comment:
Could you instead get a reference to the replica state here, e.g.:
```rust
let replica = self.compute.replicas.get_mut(&id).ok_or_else(|| ComputeError::ReplicaMissing(id))?;
```

... and then use that instead of indexing into `self.compute.replicas` each time below? (The answer might very well be "borrow-checker says no".)

Reply:
Indeed, borrow checker is happy with it, see new push!